### PR TITLE
Join sources on single destination

### PR DESCRIPTION
**Pull request by AndreasArvidsson**

### CODE REVIEW

**Review by pokey — COMMENTED** [review]
Tbh I find the impl a bit confusing, but I'm going to rework this code soon anyways so happy to leave it as long as it has decent test coverage.  But would be good to add tests for the following:
- A couple "before" examples
- Another delimited scope type, eg "arg"

**Comment by AndreasArvidsson** [comment]
The only thing I changed was to join the text on multiple sources into one destination
Nothing in this pr has changed behavior of before and after. We could probably add a couple of tests anyway.

**Comment by pokey** [comment]
> The only thing I changed was to join the text on multiple sources into one destination Nothing in this pr has changed behavior of before and after. We could probably add a couple of tests anyway.

Yeah I mean adding tests where you bring multiple before.  I could see that getting broken somehow where we end up adding two delimiters.  And same adding tests where we bring multiple to a target that is an arg.  Also prob useful "bring air and bat to arg cap"; I think currently your tests just do after and cursor

**Comment by AndreasArvidsson** [comment]
I understand in completely agree. Tests added

File: src/test/suite/fixtures/recorded/actions/bringAirAndBatAndCapToAfterItemEach.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    default.e:
+      start: {line: 4, character: 16}
+      end: {line: 4, character: 17}
+finalState:
+  documentContents: |
+    a
+    b
+    c
+
+    const values = [e, a, b, c]
```
Comment (pokey) [inline src/test/suite/fixtures/recorded/actions/bringAirAndBatAndCapToAfterItemEach.yml:48]: this is actually pretty cool 😊

**Review by pokey — APPROVED** [review]
This is great.  I get the impression these extra tests might have found a bug 😊

File: src/test/suite/fixtures/recorded/actions/bringAirAndBatAndCapToAfterItemEach.yml (hunk 1 above)
Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/actions/bringAirAndBatAndCapToAfterItemEach.yml:48]: > This is great. I get the impression these extra tests might have found a bug 😊

Maybe

Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/actions/bringAirAndBatAndCapToAfterItemEach.yml:48]: It is really cool!

**Pull request merged**